src/components/i18n/index.ts

```diff
@@ -0,0 +1,59 @@
+import defaultDictionary from './locales/en/messages.json';
+import * as _ from '../utils';
+
+/**
+ * This class will responsible for the translation through the language dictionary
+ */
+export default class I18n {
+  /**
+   * Property that stores messages dictionary
+   */
+  private static currentDictionary: {[key: string]: object} = defaultDictionary;
+
+  /**
+   * Perform translation of the string by namespace and a key
+   * If there is no translation found, returns passed key as a translated message
+   *
+   * @param namespace - path to translated string in dictionary
+   * @param dictKey - dictionary key. Better to use default locale original text
+   */
+  public static t(namespace: string, dictKey: string): string {
+    const section = I18n.getSection(namespace);
+
+    if (_.isEmpty(section)) {
+      _.log('I18n: section %o was not found in current dictionary', 'warn', namespace);
+    }
+
+    if (!section || !section[dictKey]) {
+      return dictKey;
+    }
+
+    return section[dictKey];
+  }
+
+  /**
+   * Adjust module for using external dictionary
+   *
+   * @param dictionary - new messages list to override default
+   */
+  public static setDictionary(dictionary: {[key: string]: any}): void {
```

Comment:
Type should be moved to `type` variable or interface.
Here it's `{[key: string]: any}`, however I18n.currentDictionary is `{[key: string]: object}`

Also, it's better to specify the type is recurrent:
```
type DictValue = {[string: key]: Dictionary | string} | string;
type Dictionary = {[string: key]: DictValue};
```

Reply:
However, if the root level of the dictionary has  known properties, it's better to specify them:

```
interface I18nDict {
  tools: Dictionary;
  blockTunes: Dictionary;
  ...
}
```